### DISCUSSION TITLE
Can Keycloakify app read Cookies

### DISCUSSION THREAD

**Discussion by zlatkojosic** [keycloakify/keycloakify#446] [General]
Can Keycloakify app read Cookies? Does Keycloakify app run inside Keycloak server and can it read Cookies if I install react-cookies library inside keycloakify application?

**Comment by garronej** [comment]
Hi @zlatkojosic,  
No, Keycloakify does not involves any server side rendering. You can read cookies sure but not the one from the http request, only the client side cookies.  (`cookieStore.getAll().then(console.log)`).  
This is not a limitation of Keycloakify however, Keycloak does not expose the http headers in the FreeMarker context.  
The only reliable way to share information between the apps and the theme is by adding query params to the url when redirecting to the login page.
